pydantic/_internal/_generate_schema.py

```diff
@@ -1404,12 +1404,37 @@ def _type_alias_type_schema(
             if maybe_schema is not None:
                 return maybe_schema
 
-            origin = get_origin(obj) or obj
-
-            annotation = origin.__value__
-            typevars_map = get_standard_typevars_map(obj)
+            if sys.version_info >= (3, 12):
```

Comment:
This is a bit nasty, but I haven't figured out how to reshape the logic so that there is a unified implementation for all version

Reply:
I think it's totally fine to have this branch 👍